File: closet/closetInBST.py
def cloestBST(root, target):
    res = root.key
    while root is not None:
        if root.key == target:
            return root.key
        else:
            if(abs(root.key) - target) < abs(res - target):
                res = root.key
            if root.key < target:
                root = root.right
            else:
                root = root.right
    return res

File: closet/binarySearch/closet0/closetInBST.java
public int cloest(TreeNode root, int target){
  int res = root.key;
  while(root != null){
    if(root.key == target){
      return root.key;
    }else{
      //update min_diff if necessary
      if(Math.abs(root.key - target) < Math.abs(res - target)){
        res = root.key;
      }
      //move cur pointer node 
      if(root.key < target){
        root = root.right;
      }else{
        root = root.left;
      }
    }
  }
  return res;
}
